### PR TITLE
🧮 Ticket counts

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/msgs/models.py [hunk 1]
```diff
@@ -872,16 +872,6 @@ def __str__(self):  # pragma: needs cover
         return f"BroadcastMsgCount[{self.broadcast_id}:{self.count}]"
 
 
-STOP_WORDS = (
```
Comment (rowanseymour) [inline temba/msgs/models.py:875]: sorry totally unrelated just noticed this isn't used

File: temba/tickets/models.py [hunk 2]
```diff
@@ -353,3 +354,69 @@ def get_queryset(self, org, user):
...
+            """ % {
+                "table": cls._meta.db_table
+            }
+
+            params = (distinct_set.org_id, distinct_set.status) * 2
+
+        return sql, params
+
+    @classmethod
+    def get_by_assignees(cls, org, assignees: list, status: str) -> dict:
```
Comment (rowanseymour) [inline temba/tickets/models.py:401]: for now we'll be using this with `assignees=[self.request.user, None]` for _Mine_ and _Unassigned_.. but it's optimized for potentially larger sets of users

File: temba/tickets/models.py [hunk 3]
```diff
@@ -353,3 +354,69 @@ def get_queryset(self, org, user):
...
+        Gets counts for a set of assignees (None means no assignee)
+        """
+        counts = cls.objects.filter(org=org, status=status)
+        counts = counts.values_list("assignee").annotate(count_sum=Sum("count"))
+        counts_by_assignee = {c[0]: c[1] for c in counts}
+
+        return {a: counts_by_assignee.get(a.id if a else None, 0) for a in assignees}
+
+    @classmethod
+    def get_all(cls, org, status: str) -> int:
```
Comment (rowanseymour) [inline temba/tickets/models.py:412]: for the _All_ folder which is coming in https://github.com/nyaruka/rapidpro/pull/3445

**Review by ericnewcomer — APPROVED** [review]

File: temba/tickets/tasks.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from temba.utils.celery import nonoverlapping_task
+
+from .models import TicketCount
+
+
+@nonoverlapping_task(track_started=True, name="squash_ticketcounts", lock_timeout=7200)
+def squash_ticketcounts():
```
Comment (norkans7) [inline temba/tickets/tasks.py:7]: This should be added to settings as well?

Comment (rowanseymour) [reply temba/tickets/tasks.py:7]: Oh good spot!

**Pull request merged**